src/entities/Balance/utils.ts

```diff
@@ -0,0 +1,161 @@
+      return true
+    } else {
+      logger.error('Error while fetching token balance', {
+        wallet: wallet,
+        token: balance.contractAddress,
+        error: balance.error
+      })
+      return false
+    }
+  });
+}
+
+export async function getBlockNumber(chainId: ChainId) {
+  const blockNumber = await Alchemy.get(chainId).getBlockNumber()
+  return blockNumber.result
+}
+
+export async function latestConsistentBlockNumber(chainId: ChainId) {
+  let currentBlockNumber = await getBlockNumber(chainId)
+  return (asNumber(currentBlockNumber) - asNumber(0x100)).toString(16)
+}
+
+export async function getBalances() {
+  const createdBalances = []
+  const errors = []
+
+  const wallets: WalletAttributes[] = await WalletModel.find()
+
+  for (const wallet of wallets) {
+    const tokenBalances: TokenBalanceResponse[] = await getTokenBalancesFor(wallet)
+    for (const tokenBalance of tokenBalances) {
+      try {
+        const token = await TokenModel.findMatchingToken(wallet.network, tokenBalance.contractAddress)
+        const previousBalance = await BalanceModel.findLatest(wallet.id, token.id)
+        if (previousBalance && previousBalance.amount === tokenBalance.tokenBalance) {
+          return
+        } else {
+          const newBalance: BalanceAttributes = {
+            id: uuid(),
+            wallet_id: wallet.id,
+            token_id: token.id,
+            amount: tokenBalance.tokenBalance!,
+            created_at: Time.utc().toJSON() as any
+          }
+          await BalanceModel.create(newBalance)
+          createdBalances.push(newBalance)
+        }
+      } catch (err) {
+        errors.push({ wallet: wallet, balance: tokenBalance, error: err.message })
+      }
+    }
+  }
+
+  if (errors.length > 0) {
+    logger.error('Errors while creating balances: ', { errors: errors })
+  }
+
+  return createdBalances
+}
+
+function alphabeticalSort(a:string, b:string) {
+  return ('' + a).localeCompare(b)
+}
+
+function balanceSort(a:TokenBalance, b:TokenBalance) {
+    const amountA = formattedTokenBalance(a)
+    const amountB = formattedTokenBalance(b)
+  if (amountA > amountB) {
+      return -1;
+  }
+  if (amountA < amountB) {
+    return 1;
+  }
+  return 0;
+}
+
+export async function aggregateBalances(latestBalances: BalanceAttributes[]):Promise<AggregatedTokenBalance[]> {
+  const transparencyTokens:Partial<TokenAttributes>[] = await TokenModel.getTokenList()
+  const tokenBalances:AggregatedTokenBalance[] = []
+
+  for (const token of transparencyTokens) {
+    tokenBalances.push({
+      tokenTotal: {
+        name: token.name!,
+        symbol: token.symbol!,
+        amount: toPaddedHexString(0),
+        decimals: token.decimals!
+      },
+      tokenInWallets: []
+    })
+  }
+
+  for (const balance of latestBalances) {
+    const token: TokenAttributes = await TokenModel.findToken(balance.token_id)
+    const wallet = await WalletModel.findWallet(balance.wallet_id)
```

Comment:
same here,  it would be better to query the database just once with the list of wallet